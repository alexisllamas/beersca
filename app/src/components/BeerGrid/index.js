import React from 'react';
import BeerCard from '../BeerCard';

const BeerGrid = ({ beers }) => {
  const beerCards = beers.map(beer => <BeerCard key={beer.id} {...beer} />);
  return <div className="columns is-multiline beers">{beerCards}</div>;
};

export default BeerGrid;
