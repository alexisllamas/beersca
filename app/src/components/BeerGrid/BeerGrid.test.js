import React from 'react';
import mount from 'enzyme/mount';
import BeerGrid from '.';
import NoBeerFounded from '../NotBeersFounded';

describe('BeerGrid component', () => {
  const beers = [
    {
      id: 1,
      name: 'Cerveza de Alexis',
      abv: 10,
      description: 'Cerveza bonita uwu',
      brewers_tips: 'tómala fría',
      image_url: 'img',
    },
  ];
  const wrapper = mount(<BeerGrid beers={beers} />);
  it('should have beerCards', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should not have beerCards when empty', () => {
    wrapper.setProps({ beers: [] });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toContainReact(<NoBeerFounded />);
  });
});
