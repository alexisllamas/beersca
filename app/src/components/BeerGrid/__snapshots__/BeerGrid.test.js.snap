// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerGrid component should have beerCards 1`] = `
<BeerGrid
  beers={
    Array [
      Object {
        "abv": 10,
        "brewers_tips": "tómala fría",
        "description": "Cerveza bonita uwu",
        "id": 1,
        "image_url": "img",
        "name": "Cerveza de Alexis",
      },
    ]
  }
>
  <div
    className="columns is-multiline beers"
  >
    <BeerCard
      abv={10}
      brewers_tips="tómala fría"
      description="Cerveza bonita uwu"
      id={1}
      image_url="img"
      key="1"
      name="Cerveza de Alexis"
    >
      <div
        className="card column beer is-half"
      >
        <header
          className="card-header"
        >
          <p
            className="card-header-title"
          >
            Cerveza de Alexis
          </p>
          <span
            className="tag is-danger"
          >
            10
            %
          </span>
        </header>
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            Cerveza bonita uwu
          </div>
          <div
            className="notification"
          >
            tómala fría
          </div>
          <img
            alt="the beer"
            className="beer-image"
            src="img"
          />
        </div>
      </div>
    </BeerCard>
  </div>
</BeerGrid>
`;

exports[`BeerGrid component should not have beerCards when empty 1`] = `
<BeerGrid
  beers={Array []}
>
  <NotBeersFounded>
    <div
      className="notification is-danger"
    >
      Sorry, we found no beers :(
    </div>
  </NotBeersFounded>
</BeerGrid>
`;
